crates/uv-install-wheel/src/linker.rs

```diff
@@ -497,6 +551,18 @@ fn synchronized_copy(from: &Path, to: &Path, locks: &Locks) -> std::io::Result<(
     Ok(())
 }
 
+/// Warn when a module exists in multiple packages.
+fn warn_module_conflict(locks: &Locks, filename: &WheelFilename, relative: &Path) {
+    // Check for `__init__.py` to account for namespace packages.
+    // TODO(konsti): We need to warn for overlapping namespace packages, too.
+    if relative.components().count() == 2
+        && relative.components().next_back().unwrap().as_os_str() == "__init__.py"
+    {
+        // Modules must be UTF-8, but we can skip the conversion using OsStr.
+        locks.warn_module_conflict(relative.components().next().unwrap().as_os_str(), filename);
```

Comment:
What does this look like for modules like `foo.bar`? Are we only going to show `bar`? Or are you only checking for the top-level modules because of `relative.components.count() == 2`?